File: Data Structures & Algorithms/contains-duplicate-ii/submission-9.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        if k == 0:
            return False
        window = set()
        left = 0
        right = left
        while right < len(nums):
            if right - left > k:
                window.remove(nums[left])
                left += 1
            if nums[right] in window:
                return True
            window.add(nums[right])
            right += 1
        return False

File: Data Structures & Algorithms/binary-search/submission-9.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        function helper(left, right) {
            let mid = Math.floor((left + right) / 2)
            if(left > right) return -1
            
            if (target === nums[mid]) return mid;
            else if (target < nums[mid]) return helper(left, mid - 1)
            else return helper(mid + 1, right)
        }
        return helper(0, nums.length - 1)
    }
}
